src/app/wallet/view.nim

```diff
@@ -265,35 +265,41 @@ QtObject:
     notify = accountListChanged
   
   proc estimateGas*(self: WalletView, from_addr: string, to: string, assetAddress: string, value: string): string {.slot.} =
-    try:
-      var response: int
-      if assetAddress != ZERO_ADDRESS and not assetAddress.isEmptyOrWhitespace:
-        response = self.status.wallet.estimateTokenGas(from_addr, to, assetAddress, value)
-      else:
-        response = self.status.wallet.estimateGas(from_addr, to, value)
+    var
+      response: int
+      success: bool
+    if assetAddress != ZERO_ADDRESS and not assetAddress.isEmptyOrWhitespace:
+      response = self.status.wallet.estimateTokenGas(from_addr, to, assetAddress, value, success)
+    else:
+      response = self.status.wallet.estimateGas(from_addr, to, value, success)
+    if success:
       result = $(%* { "result": %response })
-    except RpcException as e:
-      result = $(%* { "error": %* { "message": %e.msg }})
+    else:
+      result = $(%* { "error": %* { "message": %response }})
 
   proc transactionWasSent*(self: WalletView, txResult: string) {.signal.}
 
   proc transactionSent(self: WalletView, txResult: string) {.slot.} =
     self.transactionWasSent(txResult)
 
-
-  proc sendTransaction*(self: WalletView, from_addr: string, to: string, assetAddress: string, value: string, gas: string, gasPrice: string, password: string): string {.slot.} =
-    try:
-      let wallet = self.status.wallet
-      if assetAddress != ZERO_ADDRESS and not assetAddress.isEmptyOrWhitespace:
-        spawnAndSend(self, "transactionSent") do:
-          let response = wallet.sendTokenTransaction(from_addr, to, assetAddress, value, gas, gasPrice, password)
+  proc sendTransaction*(self: WalletView, from_addr: string, to: string, assetAddress: string, value: string, gas: string, gasPrice: string, password: string) {.slot.} =
+    let wallet = self.status.wallet
+    if assetAddress != ZERO_ADDRESS and not assetAddress.isEmptyOrWhitespace:
+      spawnAndSend(self, "transactionSent") do:
+        var success: bool
+        let response = wallet.sendTokenTransaction(from_addr, to, assetAddress, value, gas, gasPrice, password, success)
+        if success:
           $(%* { "result": %response })
-      else:
-        spawnAndSend(self, "transactionSent") do:
-          let response = wallet.sendTransaction(from_addr, to, value, gas, gasPrice, password)
+        else:
+          $(%* { "error": %* { "message": %response }})
+    else:
+      spawnAndSend(self, "transactionSent") do:
+        var success: bool
+        let response = wallet.sendTransaction(from_addr, to, value, gas, gasPrice, password, success)
+        if success:
           $(%* { "result": %response })
-    except RpcException as e:
-      result = $(%* { "error": %* { "message": %e.msg }})
+        else:
+          $(%* { "error": %* { "message": %response }})
```

Comment:
I tried combining these by including the `assetAddress` condition inside of `spawnAndSend` but there a very hard-to-debug error relating to the expression inside of the `do:`. This was much easier than trying to break that one down haha.

Reply:
Yeah I tried that too, but I couldn't make it work at all either.